src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1997,11 +1997,14 @@ JvmtiHandshake::execute(JvmtiUnitedHandshakeClosure* hs_cl, jthread target) {
 void
 JvmtiHandshake::execute(JvmtiUnitedHandshakeClosure* hs_cl, ThreadsListHandle* tlh,
                         JavaThread* target_jt, Handle target_h) {
+  bool is_virtual = java_lang_VirtualThread::is_instance(target_h());
   bool self = target_jt == JavaThread::current();
 
-  hs_cl->set_self(self);           // needed when suspend is required for non-current target thread
+  hs_cl->set_target_jt(target_jt);   // can be needed in the virtual thread case
+  hs_cl->set_is_virtual(is_virtual); // needed when suspend is required for non-current target thread
```

Comment:
Is this comment correct? Seems like it was just copied.

Reply:
Good catch. It was a copy-paste of the wrong comment. Will fix it.